src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs

```diff
@@ -2524,6 +2524,64 @@ public class Goo : IGoo
                 """);
         }
 
+        public interface IGooFromMetadata
+        {
+            void BarMethod();
+            int BarProp { get; }
+            int this[int x] { get; }
+            event System.Action E;
+        }
+
+        public class GooFromMetadata : IGooFromMetadata
+        {
+            void IGooFromMetadata.BarMethod() { }
+            int IGooFromMetadata.BarProp => 0;
+            int IGooFromMetadata.this[int x] => 0;
+            event System.Action IGooFromMetadata.E { add { } remove { } }
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66375")]
+        public void TestExplicitInterface2()
+        {
+            var compilation = _emptyCompilation.AddReferences(TestMetadata.Net451.mscorlib, TestMetadata.Net451.System, TestMetadata.Net451.SystemCore, TestMetadata.Net451.SystemRuntime, TestReferences.NetFx.ValueTuple.tuplelib)
+                .AddReferences(MetadataReference.CreateFromFile(typeof(GooFromMetadata).Assembly.Location));
+
+            var type = compilation.GetTypeByMetadataName(typeof(GooFromMetadata).FullName);
+            var method = type.GetMembers().Single(m => m is IMethodSymbol method && method.MethodKind == MethodKind.ExplicitInterfaceImplementation &&
```

Comment:
Note how these differ from the test above `TestExplicitInterface1` that runs from source symbols.

Reply:
Yes, it looks like `IMethodSymbol.MethodKind` for accessors of an explicit indexer is inconsistent for source and metadata: from source, `MethodKind` is `PropertyGet`; from metadata, `MethodKind` is `ExplicitImplementation`.